Code/Framework/AzToolsFramework/Tests/SpinBoxTests.cpp

```diff
@@ -293,31 +296,44 @@ namespace UnitTest
 
     TEST_F(SpinBoxFixture, SpinBoxCheckHighValueTruncatesCorrectly)
     {
-        QString value = setupTruncationTest("0.9999999");
+        QLocale testLocale{ QLocale() };
+        QString testString = "0" + QString(testLocale.decimalPoint()) + "9999999";
+        QString value = setupTruncationTest(testString);
 
-        EXPECT_TRUE(value == "0.999");
+        testString = "0" + QString(testLocale.decimalPoint()) + "999";
+        EXPECT_TRUE(value == testString);
```

Comment:
It is better to use EXPECT_STREQ instead of EXPECT_TRUE as it provides a better error message.
It would indicate the value of both variables, as opposed to `EXPECT_TRUE`
```suggestion
        EXPECT_STREQ(testString.toUtf8().constData(), value.toUtf8().constData());
```